src/components/ListItem.jsx

```diff
@@ -14,24 +20,54 @@ export function ListItem({ listPath, item, name }) {
 	useEffect(() => {
 		if (item.dateLastPurchased) {
 			const isExpired = itemIsExpired(item);
-
 			if (isExpired) {
 				updateItem(listPath, item, true);
 			}
 
 			setIsChecked(!isExpired);
 		}
-	}, [item.dateLastPurchased]);
+	}, [item]);
+
+	const getPurchaseUrgency = (item) => {
+		const today = Timestamp.now();
+		const numberOfDays = getDaysBetweenDates(today, item.dateNextPurchased);
+
+		const isInactive = itemIsInactive(item);
+		const isOverdue = itemIsOverdue(item);
+		const daysPastDue =
+			isOverdue && getDaysBetweenDates(item.dateNextPurchased, today);
+
+		switch (true) {
+			case isInactive:
+				return 'inactive (last purchased over 60 days ago)';
+			case isOverdue:
+				return `overdue (${daysPastDue === 1 ? '1 day past' : `${daysPastDue} days past`})`;
+			case numberOfDays <= 7:
+				return `soon (in ${numberOfDays === 1 ? '1 day' : `${numberOfDays} days`})`;
+			case numberOfDays >= 7 && numberOfDays < 30:
```

Comment:
@sillytsundere I think you're right. If the number of days is 7 it would always hit the first case but never the second.. so your change makes sense. Thanks!!

Reply:
Fixed!